Bindings/Java/OpenSimJNI/OpenSimContext.cpp

```diff
@@ -450,7 +451,16 @@ void OpenSimContext::setSocketConnecteeName(AbstractSocket& socket,
     clonesocket.setConnecteeName(componentPathName);
     // The following line either succeeds or throws, if the latter happens then
     // neither model or socket are changed and the message will be caught by GUI
-    clonedModel->initSystem();
+    try {
+        clonedModel->initSystem();
+    }
+    catch(const std::exception& ex) {
+        std::string message = "Unable to connect Socket<"
+            + socket.getConnecteeTypeName() + "> '" + socket.getName() +
+            "' to Component at '" + componentPathName + "'.\n Reason:" +
+            ex.what();
+        OPENSIM_THROW(Exception, message);
+    }
```

Comment:
@aymanhab I added the `try-catch` block that I was expecting.  I think it is useful that the user get a message about what just failed and then get details about why it failed instead of just the low-level details alone. Let me know if you agree.

Reply:
Thanks for making the change @aseth1 I agree. 
Generally I try to avoid having the OpenSimContext class handle exceptions but rather pass them through since this goes out of sync quickly if the API throws exceptions for other reasons, causing messages to be misleading or incorrect. This is a special case since the failure happens way downstream from the change. Keep in mind that scripting users will see the low-level error message anyway rather than the beautified version so we should continue to improve these regardless.